### PR TITLE
bpo-32843: Additional changes to test.support docs

### PR DESCRIPTION
**Pull request by csabella**
https://bugs.python.org/issue32843

### CODE REVIEW

File: Doc/library/test.rst [hunk 1]
```diff
@@ -691,7 +687,10 @@ The :mod:`test.support` module defines the following functions:
 
 .. function:: disable_faulthandler()
 
-   A context manager that replaces ``sys.stderr`` with ``sys.__stderr__``.
+   A context manager that uses ``sys.__stderr__`` instead of
+   :data:`sys.stderr`, since :mod:`test.regrtest` replaces :data:`sys.stderr`
+   with a :class:`io.StringIO` which has no file descriptor when a test is run
+   with ``-W/--verbose3``.
```
Comment (vstinner) [inline Doc/library/test.rst:693]: Well, no. The purpose of this context manager is to disable temporarily faulthandler.

**Review by vstinner — APPROVED** [review]
LGTM. Just the disable_faulthandler() doc is wrong.

**Comment by matrixise** [comment]
@csabella Could you update your PR? Thanks

File: Doc/library/test.rst [hunk 2]
```diff
@@ -249,7 +265,11 @@ The :mod:`test.support` module defines the following constants:
 
 .. data:: FS_NONASCII
 
-   A non-ASCII character encodable by :func:`os.fsencode`.
+   A non-ASCII character encodable by :func:`os.fsencode`.  The function
```
Comment (JulienPalard) [inline Doc/library/test.rst:268]: "The function" may be a bit opaque here. Maybe replace `The function` by `Choosen at import time` (which breaks the rest of the sentence)?

File: Doc/library/test.rst [hunk 3]
```diff
@@ -260,7 +280,7 @@ The :mod:`test.support` module defines the following constants:
 
 .. data:: TESTFN_UNICODE
 
-    Set to a non-ASCII name for a temporary file.
+    Set to a non-ASCII filename containing several unicode characters.
```
Comment (JulienPalard) [inline Doc/library/test.rst:283]: Looks redundent to me (*non-ASCII* and *several unicode*). Also an "unicode character" may be one in the ASCII table.

Maybe "Set to a non-ASCII filename." is enough? In contrast this one is *not* tested against being fsencodable.

File: Doc/library/test.rst [hunk 4]
```diff
@@ -1238,21 +1229,21 @@ The :mod:`test.support` module defines the following classes:
...
 
 
 .. class:: DirsOnSysPath(*paths)
 
-   A context manager to temporarily add directories to sys.path.
+   A context manager to temporarily add directories to :data:`sys.path`.
 
-   This makes a copy of :data:`sys.path`, appends any directories given
-   as positional arguments, then reverts :data:`sys.path` to the copied
+   This makes a copy of ``sys.path``, appends any directories given
```
Comment (JulienPalard) [inline Doc/library/test.rst:1242]: ```:data:`sys.path` ```  was creating a link to it, ``` ``sys.path`` ``` won't, why changing it? (multiple occurrences following)

**Review by JulienPalard — CHANGES_REQUESTED** [review]

**Comment by JulienPalard** [comment]
@csabella Hi Cheryl, long time no see, what about this PR? It accumulated a few conflicts :-(

File: Doc/library/test.rst [hunk 5]
```diff
@@ -225,6 +225,22 @@ This module defines the following exceptions:
 
 The :mod:`test.support` module defines the following constants:
 
+
+.. data:: HOST
```
Comment (JelleZijlstra) [inline Doc/library/test.rst:229]: This no longer exists on main

File: Doc/library/test.rst (hunk 2 above)
Comment (JelleZijlstra) [reply Doc/library/test.rst:268]: I think the extra text is unnecessary here; we don't need to explain how the object is derived, only what it's for.

File: Doc/library/test.rst [hunk 6]
```diff
@@ -334,12 +351,6 @@ The :mod:`test.support` module defines the following constants:
    Set to :data:`sys.maxsize` for big memory tests.
 
 
-.. data:: max_memuse
```
Comment (JelleZijlstra) [inline Doc/library/test.rst:337]: I think it's better to keep this documented for symmetry with `real_max_memuse`.

File: Doc/library/test.rst [hunk 7]
```diff
@@ -592,17 +588,17 @@ The :mod:`test.support` module defines the following functions:
...
 
-.. function:: get_original_stdout
+.. function:: get_original_stdout()
 
    Return the original stdout set by :func:`record_original_stdout` or
    ``sys.stdout`` if it's not set.
 
 
-.. function:: strip_python_strerr(stderr)
+.. function:: strip_python_stderr(stderr)
```
Comment (JelleZijlstra) [inline Doc/library/test.rst:597]: This was removed in 3.9.0a2

**Review by JelleZijlstra — COMMENTED** [review]
This unfortunately has a lot of conflicts now, enough that it's easier to start from scratch. I'll close this and open a new

**Pull request closed without merging**